okio/src/main/java/okio/AsyncTimeout.java

```diff
@@ -79,12 +77,19 @@ public final void enter() {
     scheduleTimeout(this, timeoutNanos, hasDeadline);
   }
 
-  private static synchronized void scheduleTimeout(
+  private void scheduleTimeout(
       AsyncTimeout node, long timeoutNanos, boolean hasDeadline) {
     // Start the watchdog thread and create the head node when the first timeout is scheduled.
-    if (head == null) {
-      head = new AsyncTimeout();
-      new Watchdog().start();
+    // Double-check lock to initialize the watchdog thread.
+    if (queue == null) {
+      synchronized (watchdogLock) {
+        if (queue == null) {
+          queue = new DelayQueue<AsyncTimeout>();
+          final Watchdog watchdog = new Watchdog();
+          watchdog.setPriority(Thread.MIN_PRIORITY);
```

Comment:
On Android the watchdog is running at the same priority as the main/UI thread, while it really should be running as a background thread. Is there a value between min/default that I should use? Otherwise min seems to be the best choice here.